rules/0.17-allow-new-players.py

```diff
@@ -1,10 +1,11 @@
-import util
-
 # This value set to 0 until we figure out a good way to prevent dummy account abuse
 max_start_bonus = 0
 
 
 def should_allow(pr):
+    if not pr.creates_bonus():
+        return True
```

Comment:
Returning true here means that any PR that isn't creating bonuses is allowed to merge without checking any other rules. Is that what you intended?

Reply:
Ah you're right, I had misunderstood `determine_if_mergeable` to require all rules to pass for a PR to be mergeable. It sounds like the intention is instead for a diff to be acceptable if *any* rule passes? And fail if *any* blocking rule raises an exception?

That doesn't make a ton of sense intuitively to me.

Reply:
It's acceptable if any _allow_ rule passes (by not throwing), and fails if any block rule fails (by throwing).

The reason is that we want to be able to say, e.g., "if this is a points transfer we don't need to check for vetoes".

Reply:
The idea is that there are two kinds of rules, "allow if" rules and "block if" rules.  A rule can either say "yes, trigger" in which case the PR is allowed/blocked and processing stops, or it can say "no opinion" in which case processing continues with the next rule.

Reply:
How is "no opinion" indicated? `return False`? Perhaps the right answer here.

Reply:
In the case of allow rules, no opinion is throwing an exception indicating why the PR wasn't allowed by that rule.